### PR TITLE
Added disablePhantomSpawningInMushroomFields rule.

### PR DESCRIPTION
**Pull request by Serzbiedry**
Added new rule

### CODE REVIEW

File: src/main/java/gillycarpetaddons/GillyCarpetAddonsSettings.java [hunk 1]
```diff
@@ -35,4 +35,7 @@ public class GillyCarpetAddonsSettings {
 
     @Rule(categories = {FEATURE, GILLY})
     public static boolean phantomsObeyHostileMobCap = false;
+
+    @Rule(categories = {FEATURE, GILLY})
```
Comment (Gilly7CE) [inline src/main/java/gillycarpetaddons/GillyCarpetAddonsSettings.java:39]: Same here for alphabetical order.

File: src/main/java/gillycarpetaddons/mixins/PhantomSpawner_PhantomsObeyHostileMobCapMixin.java
Comment (Gilly7CE) [inline src/main/java/gillycarpetaddons/mixins/PhantomSpawner_PhantomsObeyHostileMobCapMixin.java:1]: Please rename this file and class `PhantomSpawnerMixin` to reflect this is now a generic mixin for `PhantomSpawner`.

File: src/main/resources/assets/gilly7ce-carpet-addons/lang/en_us.json [hunk 2]
```diff
@@ -8,5 +8,6 @@
   "carpet.rule.netheritePickaxeInstantMineDeepslate.desc": "A netherite pickaxe with efficiency V combined with the haste II status effect will instant mine deepslate type blocks.",
   "carpet.rule.netheritePickaxeInstantMineEndStone.desc": "A netherite pickaxe with efficiency V combined with the haste II status effect will instant mine end stone type blocks.",
   "carpet.rule.netheritePickaxeInstantMineNetherBricks.desc": "A netherite pickaxe with efficiency V combined with the haste II status effect will instant mine nether brick type blocks.",
-  "carpet.rule.phantomsObeyHostileMobCap.desc": "Phantoms will no longer spawn if the hostile mobcap is full. This is per player."
+  "carpet.rule.phantomsObeyHostileMobCap.desc": "Phantoms will no longer spawn if the hostile mobcap is full. This is per player.",
+  "carpet.rule.phantomObeyMushroomBiome.desc": "Phantoms will no longer spawn in a mushroom fields biome."
```
Comment (Gilly7CE) [inline src/main/resources/assets/gilly7ce-carpet-addons/lang/en_us.json:12]: rename from `carpet.rule.phantomObeyMushroomBiome.desc` to `carpet.rule.disablePhantomSpawningInMushroomFields.desc` and move to top where it is in alphabetical order

**Review by Gilly7CE — CHANGES_REQUESTED** [review]
All very good! Thanks for these changes. 

Just 1 small mistake in `en_us.json` and for the rules to be ordered alphabetically.

File: src/main/java/gillycarpetaddons/mixins/PhantomSpawner_PhantomsObeyHostileMobCapMixin.java [hunk 3]
```diff
@@ -52,4 +57,20 @@ private boolean isSpectatorOrAtMobCap(PlayerEntity instance) {
...
+
+    @Redirect(
+            method = "spawn",
+            at = @At(
+                    value = "INVOKE",
+                    target = "Lnet/minecraft/server/world/ServerWorld;spawnEntityAndPassengers(Lnet/minecraft/entity/Entity;)V"
+            )
+    )
+    private void isInMooshromBiome(ServerWorld instance, Entity entity) {
+        System.out.println("bruh");
```
[the remark this answers is not in the document]
Comment (Serzbiedry) [reply src/main/java/gillycarpetaddons/mixins/PhantomSpawner_PhantomsObeyHostileMobCapMixin.java:69]: bruh thank God it didn't make it to production

**Pull request merged**